.changes/breaking/2887.md

```diff
@@ -0,0 +1,4 @@
+Bump Rust version to `1.85.0`
+
+Also added a step using `wasm-tools` when building the `wasm-executor`.
+We use it to ensure the WASM binaries have canonical LEB128 encodings.
```

Comment:
It forces everyone who uses `fuel-core` with `wasm-executor` feature to install additional dependency. We don't want to force people to install something that potentially could be avoided.